Cython/Utility/ObjectHandling.c

```diff
@@ -2205,14 +2205,14 @@ static CYTHON_INLINE PyObject *__Pyx_CallUnboundCMethod2(__Pyx_CachedCFunction *
         PyObject *args[2] = {arg1, arg2};
         if (cfunc->flag == METH_FASTCALL) {
             #if PY_VERSION_HEX >= 0x030700A0
-            return (*(__Pyx_PyCFunctionFast)(void*)(PyCFunction)cfunc->func)(self, args, 2);
+            return (*(__Pyx_PyCFunctionFast)(void(*)(void))(PyCFunction)cfunc->func)(self, args, 2);
```

Comment:
This gets verbose and difficult to read. Since we use macros here already, what do you think of adding a macro for making the casts and the call in one go?
```c
__Pyx_call_PyCFunctionFast(func, args, nargs)
```

Reply:
@scoder I targeted this PR to branch 3.0.x. If you think such refactors are appropriate for the release branch, then sure, we can do it. Otherwise, we should make such change in another PR targeting master once this one get merged in. Please confirm your preference.

Reply:
@scoder I added a commit on top implementing the change you suggested, albeit with a slightly different naming convention `__Pyx_call_PyCFunction` ->  `__Pyx_CallCFunction`. The main rationale for this choice is that the `cfunc` argument is not really a `PyCFunction` (function) pointer, but a pointer to `struct __Pyx_CachedCFunction`.